aiida/plugins/entry_point.py

```diff
@@ -268,6 +285,37 @@ def get_entry_point(group: str, name: str) -> EntryPoint:
     return found[name]
 
 
+def convert_potentially_deprecated_entry_point(group: str, name: str) -> str:
+    """Check whether the specified entry point is deprecated, in which case print warning and convert to new name.
+
+    For `aiida-core==2.0` all existing entry points where properly prefixed with ``core.`` and the old entry points were
+    deprecated. To provide a smooth transition these deprecated entry points are detected in ``get_entry_point``, which
+    is the lowest function that tries to resolve an entry point string, by calling this function.
+
+    If the entry point corresponds to a deprecated one, a warning is raised and the new corresponding entry point name
+    is returned.
+
+    This method should be removed in ``aiida-core==3.0``
+    """
+    from warnings import warn
+
+    from aiida.common.warnings import AiidaDeprecationWarning
```

Comment:
Is there a reason to not have these as top-level imports?

Reply:
Because this function is the only place that uses them and when the function gets removed the obsolete imports are also automatically taken care over. Not a huge deal, but why not?

Reply:
Meh, pylint will flag unused imports, so that's not an issue.
Yeh not a big deal, but that's the python coding standard that we use (outside reasons like CLI response and circular imports), whereas "single use imports should be internal to functions" is not.
Up to you